nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/JCAMPData.java

```diff
@@ -17,987 +17,569 @@
     @Override
-    public void readVector(int iVec, Vec dvec) {
-        dvec.setGroupDelay(groupDelay);
-        ArrayList<Double> rValues = rparser.getYValues();
-        int n = rValues.size();
+    public void readVector(int index, Vec dvec) {
+        double[] rValues = realPages.get(index).toArray();
+        int n = rValues.length;
 
-        if (iparser == null) {
+        if (imaginaryPages.isEmpty()) {
             dvec.resize(n, false);
             for (int i = 0; i < n; i++) {
-                dvec.set(i, rValues.get(i));
+                dvec.set(i, rValues[i]);
                 dvec.setTDSize(n);
             }
         } else {
-            ArrayList<Double> iValues = iparser.getYValues();
+            double[] iValues = imaginaryPages.get(index).toArray();
             dvec.resize(n, true);
             dvec.setTDSize(n);
             for (int i = 0; i < n; i++) {
-                dvec.set(i, iValues.get(i), rValues.get(i));
+                //WARNING: real and imaginaries are inverted on purpose
+                dvec.set(i, iValues[i], rValues[i]);
             }
-
         }
 
+        dvec.setGroupDelay(getGroupDelay());
+
         dvec.dwellTime = 1.0 / getSW(0);
         dvec.centerFreq = getSF(0);
 
-        double delRef = (dvec.getSize() / 2 - 0) * (1.0 / dvec.dwellTime) / dvec.centerFreq / dvec.getSize();
+        double delRef = (dvec.getSize() / 2d) * (1.0 / dvec.dwellTime) / dvec.centerFreq / dvec.getSize();
         dvec.refValue = getRef(0) + delRef;
-        //System.out.println("zeroref " + dvec.refValue);
-        //dvec.setPh0(getPH0(1));
-        //dvec.setPh1(getPH1(1));
-    }
-
-    @Override
-    public void readVector(int iDim, int iVec, Vec dvec) {
-        int shiftAmount = 0;
-        if (groupDelay > 0) {
-            // fixme which is correct (use ceil or not)
-            //shiftAmount = (int)Math.round(Math.ceil(groupDelay));
-            shiftAmount = (int) Math.round(groupDelay);
-        }
-        if (dvec.isComplex()) {
-            if (dvec.useApache()) {
-                readVector(iDim, iVec + shiftAmount, dvec.getCvec());
-            } else {
-// fixme
-                readVector(iVec + shiftAmount, dvec.rvec, dvec.ivec);
-            }
-        } else {
-// fixme
-            readVector(iVec, dvec.rvec);
-        }
-        dvec.dwellTime = 1.0 / getSW(iDim);
-        dvec.centerFreq = getSF(iDim);
-        dvec.refValue = getRef(iDim);
-        dvec.setPh0(getPH0(iDim));
-        dvec.setPh1(getPH1(iDim));
-        if (iDim == 0) {
-            dvec.setGroupDelay(groupDelay);
-        } else {
-            dvec.setGroupDelay(0.0);
-        }
     }
 
     @Override
     public void readVector(int iVec, Complex[] cdata) {
-        byte[] dataBuf = new byte[tbytes];
-        readVecBlock(iVec, dataBuf);
-        copyVecData(dataBuf, cdata);
+        //TODO implement me? needed?
+        throw new UnsupportedOperationException("Not yet implemented: readVector(int iVec, Complex[] cdata)");
```

Comment:
All these readVector() alternatives are never called, as far as I could see while testing. If they are only be called by other NMRData implementation internally, we should remove them from the interface. Otherwise, I can implement them if it is useful.

Reply:
I should probably review all these variants of readVector.  They go way back to some original versions where there were different choices for different data file arrangements etc.  We can probably get rid of most of them, but I better look pretty carefully at this.